wic/iot2050.wks

```diff
@@ -8,6 +8,6 @@
 # COPYING.MIT file in the top-level directory.
 #
 
-part / --source rootfs-u-boot --sourceparams="no_initrd=yes,script_prepend=env exists fdtfile || if test \\\"\${board_name}\\\" = \\\"IOT2050-ADVANCED\\\"; then set fdtfile siemens/iot2050-advanced.dtb; else setenv fdtfile siemens/iot2050-basic-oldfw.dtb; fi" --fstype ext4 --label rootfs --align 1024 --use-uuid
+part / --source rootfs-u-boot --sourceparams="no_initrd=yes,script_prepend=env exists fdtfile || if test \\\"\${board_name}\\\" = \\\"IOT2050-ADVANCED\\\"; then set fdtfile siemens/iot2050-advanced-oldfw.dtb; else setenv fdtfile siemens/iot2050-basic-oldfw.dtb; fi" --fstype ext4 --label rootfs --align 1024 --use-uuid
```

Comment:
That's not yet the logic I sketched. This will select "oldfw" as dtb for the advanced board also over the new firmware. Note the `||`-

Reply:
the `||` , new firmware exists the env fdtfile , so It would not execute the following code .
I think it would set the fdtfile = iot2050-advanced.dtb not iot2050-advanced-oldfw.dtb

Reply:
You are right, the whole if-statement will not be run. All should be fine then.